examples/tutorials/voxel_framework_tutorial.py

```diff
@@ -0,0 +1,137 @@
+# [setup]
```

Comment:
This is already a very large PR. Can this tutorial be a separate PR? The problem with large PRs is that we hit reviewer fatigue and not everything gets properly reviewed.

Reply:
Yeah that makes sense - I'll split up the tutorial and python bindings into another PR.